File: python/math_utils/angle.py
import math

RADIANS_PER_DEGREE = math.pi/180

def d2r(deg):
    return deg*RADIANS_PER_DEGREE

def r2d(rad):
    return rad/RADIANS_PER_DEGREE

def normalize(rad): 
    return rad % (math.pi*2)

def radians(rad):
    return normalize(rad)

def degrees(deg):
    return normalize(d2r(deg))

def opposite(rad):
    return normalize(rad + math.pi)

def diffClockwise(a, b):
    return normalize(a - b)

def diffCounterclockwise(a, b):
    return normalize(b - a)

def diff(a, b):
    return min(diffClockwise(a, b), 
            diffCounterclockwise(a, b))
            

File: dotnet/RSExample/Math/Angle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RSExample.Math
{
    class Angle
    {
        const float RADIANS_PER_DEGREE = MathF.PI / 180;

        public static float D2R(float deg) { return deg * RADIANS_PER_DEGREE; }

        public static float R2D(float rad) { return rad / RADIANS_PER_DEGREE; }

        public static float Normalize(float rad)
        {
            return Utils.Mod(rad, MathF.PI * 2);
        }

        public static float Radians(float rad)
        {
            return Normalize(rad);
        }

        public static float Degrees(float deg)
        {
            return Normalize(D2R(deg));
        }

        public static float Opposite(float rad)
        {
            return Normalize(rad + MathF.PI);
        }

        public static float DiffClockwise(float a, float b)
        {
            return Normalize(a - b);
        }

        public static float DiffCounterclockwise(float a, float b)
        {
            return Normalize(b - a);
        }

        public static float Diff(float a, float b)
        {
            return MathF.Min(DiffClockwise(a, b), DiffCounterclockwise(a, b));
        }
    }
}
